anvil/src/eth/api.rs

```diff
@@ -792,10 +792,14 @@ impl EthApi {
             // valid rlp and then rlp decode impl of `TypedTransaction` will remove and check the
             // version byte
             let extend = rlp::encode(&data);
-            match rlp::decode::<TypedTransaction>(&extend[..]) {
+            let tx = match rlp::decode::<TypedTransaction>(&extend[..]) {
                 Ok(transaction) => transaction,
                 Err(_) => return Err(BlockchainError::FailedToDecodeSignedTransaction),
-            }
+            };
+
+            self.ensure_typed_transaction_supported(&tx)?;
```

Comment:
How about other JSON-RPC requests for sending transactions? Namely `eth_sendTransaction` and `eth_sendUnsignedTransaction`?